### PR TITLE
Feat/lyra

### PR DESCRIPTION
**Pull request by hochbruj**
https://trello.com/c/KcE5S5U8

### CODE REVIEW

File: src/services/lyra/trade.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  market: LyraOptionMarket,
+  optionType: LyraOptionType,
+  expiry: number,
+  strike: number,
+  tradeType: LyraTradeType,
+  optionAmount: BigNumber | string,
+  amountIn: BigNumber | string,
+  assetIn: string,
+  slippage = 0.5
+): Promise<any> {
```
[the remark this answers is not in the document]
Comment (hochbruj) [reply src/services/lyra/trade.ts]: done

**Comment by SeanZom** [comment]
I found almost half of the unit tests failed, do you want to fix them in this PR?

File: src/services/lyra/markets.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+export async function getBoard(
+  network: Network,
+  market: LyraOptionMarket,
+  expiry: number
+): Promise<Board> {
+  const lyra = new Lyra(lyraNetworkMap[network]);
+  const optionMarket = await lyra.market(market);
+  const filteredBoards = optionMarket
+    .liveBoards()
+    .filter(e => e.expiryTimestamp === expiry);
```
[the remark this answers is not in the document]
Comment (hochbruj) [reply src/services/lyra/markets.ts:15]: When you trade options through the SDK you need to pass an expiry timestamp and strike price, so the correct lyra option/strike will be selected. That's why the expiry needs to match the expiry of the lyra option (board) exactly.

**Comment by hochbruj** [comment]
> I found almost half of the unit tests failed, do you want to fix them in this PR?

I know that the tests in the SDK are not set up properly. When I implemented a new feature (e.g.Lyra) I just run the test cases for that component. The problem is that these tests do on-chain transactions (not a local fork), so I don't want to run all test cases every time. There are definitely ways to improve the test cases but that's a different task.

**Comment by hochbruj** [comment]
I have added selling options and covered calls, you can everything on that pool https://app.dhedge.org/manager/0x3d45e539df0f9fa6015106a968d732a8f6c2a40b

**Pull request merged**